docs/content/guides/dagster/validating_pandas_dataframes_with_dagster_types.mdx

```diff
@@ -0,0 +1,240 @@
+
+We'll outsource the validation logic to a purpose-built library: [Pandera](https://github.com/pandera-dev/pandera). Pandera provides an expressive API for defining dataframe schemas. Schemas encode data type, boundary, nullability, and other constraints on dataframes. The workhorse method of Pandera is `schema.validate(dataframe)`. If a dataframe violates the schema contraints, this method throws an error containing a detailed record of constraint violations.
+
+Below, we'll see how to use a factory to wrap any Pandera schema in a Dagster type. The resulting types can then be bound to individual inputs and outputs in a job. This integrates Pandera's powerful validation capabilities with Dagster's boilerplate-free runtime type checking.
+
+## Setup
+
+Suppose you are a data scientist working for an e-bike rental company. You've been tasked with setting up some pipelines to analyze trip data, and have been experimenting with implementing them in Dagster. You are currently looking at the distribution of trip lengths, so you've exported a slice of trips from October into a simple CSV file with 4 columns: `bike_id`, `start_time`, `end_time`, and `station` (trips must start and end from the same station):
+
+<Image
+alt="data.png"
+src="/images/guides/validating_pandas_dataframes_with_dagster_types/data.png"
+width={730}
+height={608}
+/>
+
+The first thing you want is to visualize the distribution of trip lengths, so you set up a simple Dagster job to do this. Your job has two ops: `load_trips` reads a CSV with your data, and `plot_trip_length_dist` generates a plot of the distribution using the [Holoviews](https://holoviews.org/) plotting library.
+
+```python file=/guides/validating_pandas_dataframes_with_dagster_types/job_1.py
+from dagster import AssetMaterialization, job, op
+import numpy as np
+import pandas as pd
+import holoviews as hv
+
+@op
+def load_trips():
+    return pd.read_csv(
+        "./2021-10_ebike_trips.csv",
+        parse_dates=["start_time", "end_time"],
+    )
+
+@op
+def generate_plots(trips):
+    minute_lengths = [ x.total_seconds() / 60 for x in trips.end_time - trips.start_time ]
+    edges, freqs = np.histogram(minute_lengths, 15)
+    hist = hv.Histogram((freqs, edges)).opts(width=600, xlabel='Minutes')
+    hv.save(hist, 'trip_length_hist.png', fmt='png')
+    yield AssetMaterialization(
+        asset_key="trip_dist_plot",
+        description="Distribution of trip lengths."
+    )
+    yield Output(None)
+
+@job
+def generate_trip_plots():
+    generate_plots(load_trips)
+```
+
+If we try running this job and inspect the result, we get a strange error:
+
+```
+# => ...
+# => ValueError: autodetected range of [nan, nan] is not finite
+```
+
+## Schema
+
```

Comment:
and add a subtitle "Pandera Schema" - i found this section of explaining Pandera a bit like a detour in this guide, but i guess that's already - seems like the best we could do and it's not that heavy.


spitballing [1]: how do you feel about having a very simple `type_check_fn` without introducing a third-party lib at the beginning? something like:

```
DagsterType(type_check_fn=lambda val: val is not None, ...)
```